improve authorization configuration (#866)
SAP/luigi: docs/authorization-configuration.md

# Authorization configuration

To configure authorization in Luigi, go to the `auth:` section of your project's `basicConfiguration.js` file. To see how authorization works, you can also go to the [Luigi Fiddle](https://fiddle.luigi-project.io) site and configure a sample application.

Luigi provides OpenID Connect and OAuth2 Implicit Grant authorization out of the box. The **use** key defines the active authorization provider and the **disableAutoLogin** key allows you to disable the automatic login flow that is provided by default.

```javascript
auth: {
  use: 'openIdConnect',
  openIdConnect: {
    ...
  },
  disableAutoLogin: true
}
```

You have the following authorization options:
* [OpenID Connect](#openid-connect-configuration)

>   * [Third-party cookies and silent token refresh](#third-party-cookies-and-silent-token-refresh)

Author: It is, because that's supposed to be a sub-category of OpenID connect (that's also why its heading begins with ### instead of ##)

>   * [Third-party cookies and silent token refresh](#third-party-cookies-and-silent-token-refresh)

Reviewer: Oh, right. Please ignore it then :)

It is possible to detect whether the user's browser supports the mechanism by using the script in the [`third-party-cookies`](https://github.com/SAP/luigi/tree/master/core/third-party-cookies) catalog. Deploy these files on a **different domain** than your main application and set **thirdPartyCookiesScriptLocation** to `init.html` file. During initialization, Luigi detects the cookies support and produces a warning in the console if cookies are disabled in the user's browser.                                      

When Luigi fails to renew the token and then logs the user out, it adds the following query parameters to the logout page redirect URL: `?reason=tokenExpired&thirdPartyCookies=[VALUE]`. Luigi replaces the **VALUE**  with one of the following:
- `disabled` means that third party cookies are disabled.
- `enabled` means third party cookies are supported by the browser.
- `not_checked` means that the script was not provided in **thirdPartyCookiesScriptLocation** or it could not be loaded.

> The application developer can read these parameters and set a logout page based on them.

Reviewer: ```suggestion
Use these parameters to set a logout page.
```

To avoid using `can` you could rephrase it like this, unless it alters the message you wanted to convey.

- **expirationCheckInterval** the number of seconds to pass between each check if the token is about to expire. The default value is `5` seconds.

> ## Another authorization provider

Author: You're right, I changed it 👍

> If you are using another authorization provider, you can also implement the functions below for Luigi.

Reviewer: ```suggestion
If you use a custom authorization provider, you can also implement these functions for Luigi.
```
`You can` or `you must`? If it's optional, I think a one-sentence explanation here could be useful.
